service/worker/deletenamespace/activities.go

```diff
@@ -99,6 +109,35 @@ func (a *localActivities) ValidateProtectedNamespacesActivity(_ context.Context,
 	return nil
 }
 
+func (a *localActivities) ValidateNexusEndpointsActivity(ctx context.Context, nsID namespace.ID, nsName namespace.Name) error {
+	if !a.allowDeleteNamespaceIfNexusEndpointTarget() {
+		// Prevent deletion of a namespace that is targeted by a Nexus endpoint.
+		var nextPageToken []byte
+		for {
+			resp, err := a.nexusEndpointManager.ListNexusEndpoints(ctx, &persistence.ListNexusEndpointsRequest{
+				LastKnownTableVersion: 0,
+				NextPageToken:         nextPageToken,
+				PageSize:              a.nexusEndpointListDefaultPageSize(),
+			})
+			if err != nil {
+				a.logger.Error("Unable to list Nexus endpoints from persistence.", tag.WorkflowNamespace(nsName.String()), tag.Error(err))
+				return fmt.Errorf("unable to list Nexus endpoints for namespace %s: %w", nsName, err)
+			}
+
+			for _, entry := range resp.Entries {
+				if endpointNsID := entry.GetEndpoint().GetSpec().GetTarget().GetWorker().GetNamespaceId(); endpointNsID == nsID.String() {
+					return temporal.NewNonRetryableApplicationError(fmt.Sprintf("cannot delete a namespace that is a target of a Nexus endpoint %s", entry.GetEndpoint().GetSpec().GetName()), errors.ValidationErrorErrType, nil, nil)
```

Comment:
This would need to be translated to a FailedPrecondition not an invalid argument error, can you change and test the error conversion works as expected?

Reply:
This is a very good point. There are few other errors which are also should be `FailedPrecondition`. I have another PR focused on errors clean up, coming right after this one. Let me address this there.